src/sage/rings/finite_rings/residue_field.pyx

```diff
@@ -889,30 +956,33 @@ cdef class ReductionMap(Map):
 
         EXAMPLES::
 
+            sage: # needs sage.rings.number_field
             sage: x = polygen(ZZ, 'x')
-            sage: K.<a> = NumberField(x^3 + x^2 - 2*x + 8)                              # optional - sage.rings.number_field
-            sage: F = K.factor(2)[0][0].residue_field()                                 # optional - sage.rings.number_field
-            sage: F.reduction_map()                                                     # optional - sage.rings.number_field
+            sage: K.<a> = NumberField(x^3 + x^2 - 2*x + 8)
+            sage: F = K.factor(2)[0][0].residue_field()
+            sage: F.reduction_map()
             Partially defined reduction map:
               From: Number Field in a with defining polynomial x^3 + x^2 - 2*x + 8
               To:   Residue field of Fractional ideal (-1/2*a^2 + 1/2*a - 1)
 
-            sage: K.<theta_5> = CyclotomicField(5)                                      # optional - sage.rings.number_field
-            sage: F = K.factor(7)[0][0].residue_field()                                 # optional - sage.rings.number_field
-            sage: F.reduction_map()                                                     # optional - sage.rings.number_field
+            sage: # needs sage.rings.number_field
+            sage: K.<theta_5> = CyclotomicField(5)
+            sage: F = K.factor(7)[0][0].residue_field()
+            sage: F.reduction_map()
             Partially defined reduction map:
               From: Cyclotomic Field of order 5 and degree 4
               To:   Residue field in theta_5bar of Fractional ideal (7)
 
+            sage: # needs sage.rings.finite_rings
             sage: R.<t> = GF(2)[]; P = R.ideal(t^7 + t^6 + t^5 + t^4 + 1)
             sage: k = P.residue_field()
-            sage: k.reduction_map()
+            sage: k.reduction_map()                                                     # needs sage.libs.ntl
             Partially defined reduction map:
               From: Fraction Field of
                     Univariate Polynomial Ring in t over Finite Field of size 2 (using GF2X)
               To:   Residue field in tbar of Principal ideal (t^7 + t^6 + t^5 + t^4 + 1) of
                     Univariate Polynomial Ring in t over Finite Field of size 2 (using GF2X)
-            sage: type(k)
+            sage: type(k)                                                               # needs sage.libs.linbox
             <class 'sage.rings.finite_rings.residue_field_givaro.ResidueFiniteField_givaro_with_category'>
```

Comment:
Is this right? Needs `sage.libs.linbox` to get `type(k)` but does not to create `k`?

Reply:
Ah, I see. Thanks for explaining.

Reply:
But we may handle this in a similar way as below.

Reply:
I've made the change below because that doctest is just displaying the codomain.

Here `k` has already been displayed, so the doctest seems to be intended to display a specific type.
So I'm keeping it in the current form

Reply:
OK.